slides/06-tuning-hyperparameters.qmd

```diff
@@ -176,12 +176,66 @@ glm_rec <-
   step_ns(angle, deg_free = tune("angle")) %>%
   step_ns(distance, deg_free = tune("distance")) %>%
   step_normalize(all_numeric_predictors())
+```
+
+. . .
+
+Let's `tune()` our spline terms!
+
+## Choosing tuning parameters `r hexes("recipes","parsnip", "workflows", "tune")`
+
+Let's take our previous recipe and add a few changes:
+
+```{r}
+#| code-line-numbers: "4"
 glm_spline_wflow <-
   workflow() %>%
   add_model(logistic_reg()) %>%
   add_recipe(glm_rec)
 ```
 
+
+## {background-image="https://www.tmwr.org/figures/ames-latitude-splines-1.png" background-size="contain"}
+
+:::notes
+Splines replace the existing numeric predictor with a set of columns that allow a model to emulate a flexible, nonlinear relationship.
+
+More spline terms = more "wiggly", i.e. flexibly model a nonlinear relationship
+
+How many spline terms? This is called *degrees of freedom*
+
+2 and 5 look like they underfit; 20 and 100 look like they overfit
+:::
+
+## Splines and nonlinear relationships
+
+```{r}
+#| echo: false
+#| fig-align: center
```

Comment:
```suggestion
#| fig-align: center
#| fig.width: 16
#| fig.height: 8
```

It looked a little small on my screen. This width/height is also used for the `Single decision tree` slide, does it seem good to you? The text itself is still a little small but I think that is ok.

Before:
<img width="1920" alt="Screen Shot 2022-07-23 at 6 30 40 PM" src="https://user-images.githubusercontent.com/19150088/180625052-899acd8a-b98d-4cdb-a1a1-195363f543ab.png">

After:
<img width="1920" alt="Screen Shot 2022-07-23 at 6 31 21 PM" src="https://user-images.githubusercontent.com/19150088/180625055-234e0d8a-fe84-47f0-851f-11d43af1d4d4.png">

Reply:
Yeah, looks nice 👍